### ISSUE TITLE
Nested list causes spurious repeats

### ISSUE THREAD

**Issue by d-led** [ianka/mustache.tcl#2]
Example:

```tcl
source mustache.tcl

set template {
name: {{name}} / {{#params}}{{name}}{{^last}}, {{/last}}{{/params}}
{{#params}}
    {{name}}{{#special}} -> special{{/special}}
{{/params}}
---
}

set data {
	name bla
	params {
		{name a}
		{name b special 1}
		{name c}
		{name d special 1 last 1}
	}
}

puts [::mustache::mustache $template $data]
```
&darr;

```
name: bla / a, b, c, d
    a
    b -> special
---
    c
    d -> special
---
```

note the spurious repeated `---` 

Inverting the list:

```
    {{name}}{{^special}} -> special{{/special}}
              ^
```

does not cause the spurious repeat:

```
name: bla / a, b, c, d
    a -> special
    b
    c -> special
    d
---
```

I might be mistaken. Got to check out other implementations

**Comment by d-led** [comment]
Ruby implementation gives the expected result:

```ruby
require 'mustache'

template = <<MUSTACHE
name: {{name}} / {{#params}}{{name}}{{^last}}, {{/last}}{{/params}}
{{#params}}
    {{name}}{{#special}} -> special{{/special}}
{{/params}}
---
MUSTACHE

data = {
	name: 'bla',
	params: [
		{name: 'a'},
		{name: 'b', special: 1},
		{name: 'c'},
		{name: 'd', special: 1, last: 1}
	]
}

puts Mustache.render(template, data)
```

&darr;

```
name: bla / a, b, c, d
    a
    b -> special
    c
    d -> special
---
```

**Comment by ianka** [comment]
I see this isn't correct but can we settle on a test case (as written in the test case spec) first?

**Comment by d-led** [comment]
Sure, I'll need to dig into it a bit

**Comment by ianka** [comment]
`overview: |
  Additional tests for bugs found by mustache.tcl users.
tests:
  - name: Sections - trailing text
    desc: Sections should not repeat text following them.
    data: { o: [ { x: a, i: 1 }, { x: b, i: 0 }, { x: c, i: 1 } ] }
    template: '{{#o}}{{#i}}{{x}}{{/i}}{{/o}}foo'
    expected: 'acfoo'`

**Comment by ianka** [comment]
Here's a test case which uncovers the problem:
```yaml
overview: |
  Additional tests for bugs found by mustache.tcl users.
tests:
  - name: Sections - trailing text
    desc: Sections should not repeat text following them.
    data: { o: [ { x: a, i: 1 }, { x: b, i: 0 }, { x: c, i: 1 } ] }
    template: '{{#o}}{{#i}}{{x}}{{/i}}{{/o}}foo'
    expected: 'acfoo'
```
Unfortunately, the reason for the buggy behaviour is a too simplistic approach on how to handle text trailing sections. It will take me a while to fix this.

Thanks for reporting.

**Comment by d-led** [comment]
Thanks for the reply and the test case. I've prepared a [fork](https://github.com/d-led/mustache.tcl) but didn't come round adding the test case. Feel free to use the commits (travis config or readme formatting beginnings)

**Comment by ianka** [comment]
fixed with v1.1.3.1
